crates/evm/traces/src/lib.rs

```diff
@@ -324,6 +341,14 @@ impl TraceMode {
         std::cmp::max(self, mode.into())
     }
 
+    pub fn with_state_changes(self, yes: bool) -> Self {
+        if yes {
+            std::cmp::max(self, Self::RecordStateDiff)
+        } else {
+            self
+        }
+    }
+
     pub fn with_verbosity(self, verbosiy: u8) -> Self {
         if verbosiy >= 3 {
```

Comment:
one suggestion for separate improvement @zerosnacks 
let's introduce some enum for these numbers

Reply:
added a ticket for it here: https://github.com/foundry-rs/foundry/issues/9426